architecture/Architecture.md

```diff
@@ -0,0 +1,158 @@
+# Architecture
+
+## Concepts
+
+Every...
+
+- quantity is a struct
+- quantity has an _internal_ [unit](../quantities/units/IUnit.cs) that is either
+  - [SI](../quantities/units/Si/ISiUnit.cs) based
+  - [Imperial](../quantities/units/Imperial/IImperial.cs)
+  - [other](../quantities/units/Other/IOther.cs)
+- SI quantity supports [prefixing](../quantities/prefixes/IPrefix.cs) with
+  - a decimal prefix
+  - a binary prefix
+  - or both
+- quantity represents a unique [dimension](../quantities/dimensions/IDimension.cs)
+- quantity can only be instantiated using _generic_ factory methods
+  - that are parameterized by a combination of generic prefix and unit parameters.
+  - The unit parameter is constrained by the same dimension the quantity implements
+- quantity is implicitly convertible to a Double
+- quantity supports
+  - additive operations
+  - scalar multiplicative operations
+  - comparison to quantities of the same type
+- quantity is left associative
+  - a compound expression will take the prefix and unit of the left most term
+- quantity can be converted to any other valid combination of prefix and unit
+
+Therefore, the actual underlying unit and/or prefix of a given type is an irrelevant detail of any quantity.
+
+## Public API
+
+This library was built around an API I had in mind. The API should rely heavily on generics, which would make it very easy to use and extend for an arbitrary amount of units and (metric) prefixes.
+
+The minimal API for creation looks like this:
+
+```csharp
+public struct TQuantity : IMyDimension
+{
+    public static TQuantity Create<TPrefix, TUnit>(in Double value)
```

Comment:
I have to read more about [`Contravariant`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-generic-modifier) before I can fully understand this concept.
I never used a doulbe `Where`. Cool stuff.